### DISCUSSION TITLE
can i only send data to kafka without any oap server and how to config?

### DISCUSSION THREAD

**Discussion by sourirer** [apache/skywalking#10735] [Q&A] [answered]
I wanna package by source code with skywalking-java-agent and send trace data to kafka .

I don't deploy any grpc oap server to receive the data but only wish send trace data to kafka with kafka-report-plugins.

Any help will be appreciate.

**Comment by wu-sheng** [comment, marked as answer]
We don't support that officially.
If you want to do some local change, that is your private work.

**Reply by sourirer** [reply]
thanks for reply.
so you mean in any situation, skywalking-agent need a grpc oap server to connected or else trace data can't be send?

**Reply by wu-sheng** [reply]
Yes.
SkyWalking is the whole piece, OAP and agent connected is the condition that the agent should work. This is one of the basic protect mechanisms to reduce target service unnecessary payload.

**Reply by Superskyyy** [reply]
SkyWalking agents can already report data directly to kafka, but currently you will need an oap grpc server address, if you wish to get rid of oap entirely for some reason, it will require you to do some minor changes as profiling is not supported in the kafka reporter and there could be other mechanisms that you should bypass to just using the tracing part. See https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/advanced-reporters/

**Reply by sourirer** [reply]
thank you guys. 
I will try to do some change to implement that.
